extensions/mssql/src/sqlToolsServer.ts

```diff
@@ -166,6 +167,7 @@ function getClientOptions(context: AppContext): ClientOptions {
 			SqlMigrationService.asFeature(context),
 			SqlCredentialService.asFeature(context),
 			TableDesignerFeature,
+			BlobService.asFeature(context),
```

Comment:
@kburtram @alanrenmsft So I was looking at this stuff some more and came across a couple things that made me rethink how we should do this : 

1. I found out about/was reminded that ResourceProvider exists : https://github.com/microsoft/sqltoolsservice/tree/main/src/Microsoft.SqlTools.ResourceProvider.Core 

This seems like a much better place to have code for getting the SAS so I think having the requests go there instead make sense

2. I still don't see any reason to expose this to extensions. So my thought is that instead of adding this as a feature on the registered ResourceProvider itself (like we do with CreateFirewall) this would instead just be an "internal" message that is sent and processed. Basically adding in functions to [ResourceProvider](https://github.com/microsoft/azuredatastudio/blob/9516472f1b537d6618ee26778bf339252e692a7a/extensions/mssql/src/resourceProvider/resourceProvider.ts) that would send the messages but not register them as a full-fledged "feature", since all we care about is sending the messages and then getting the response back. This would simplify the code a lot.

Any thoughts about this from you two? I figure that starting with this is good, and then later on if we decide we want to make some sort of more generic way for extensions to retrieve these keys then we can look into adding it to the extension API at that point, but that shouldn't be necessary for the time being (and likely ever)

Reply:
To follow up on this - after talking with Nemanja we decided to keep it all in STS for now. Resource Provider is probably the better place for actually creating/getting the SAS itself, but there's two parts to this - first is creating the SAS and second is saving it to the DB. And since the resource provider doesn't have all the connection info stuff that STS does that means we would have had to split it into two parts where first it creates the SAS using RP and then it saves it using STS.

Overall that's probably the more "right" way to do it, but given the specific use case for this I figured it'd be better just to keep it simple and put it all in STS for now. If we end up wanting to make this more generic/reusable later on then we can split it up then. 

(the rest of my comments about not exposing this to other extensions/implementations through the public API/registered feature still stands though and is what we've implemented here)